CoreCLR tests build error on main for arm/armel/arm64

Next error happens during tests cross build for arm/armel/arm64 on x64:
```
/home/runtime/.packages/microsoft.net.sdk.il/7.0.0-preview.4.22208.8/targets/Microsoft.NET.Sdk.IL.targets(136,5): error MSB3073: The command ""/home/runtime/artifacts/bin/coreclr/Linux.arm64.Release/ilasm" -QUIET -NOLOGO -OPTIMIZE -DLL  -OUTPUT="/home/runtime/artifacts/tests/coreclr/obj/Linux.arm64.Release/Managed/Loader/classloader/generics/ByRefLike/InvalidCSharp/InvalidCSharp.dll"  InvalidCSharp.il" exited with code 127. [/home/runtime/src/tests/Loader/classloader/generics/ByRefLike/InvalidCSharp.ilproj] [/home/runtime/src/tests/build.proj]
/home/runtime/src/tests/Common/dir.traversal.targets(25,5): error : (No message specified) [/home/runtime/src/tests/build.proj] [/home/runtime/src/tests/build.proj]
/home/runtime/src/tests/build.proj(501,5): error MSB3073: The command ""/home/runtime/dotnet.sh" msbuild /home/runtime/src/tests/build.proj /t:Build "/p:TargetArchitecture=arm64" "/p:Configuration=Release" "/p:LibrariesConfiguration=Release" "/p:TargetOS=Linux" "/p:RuntimeOS=" "/p:RuntimeFlavor=coreclr" "/p:RuntimeVariant=" "/p:CLRTestBuildAllTargets=" "/p:__TestGroupToBuild=3" "/p:__SkipRestorePackages=1" /nodeReuse:false /maxcpucount /bl:/home/runtime/artifacts//log/Release/InnerManagedTestBuild.3.binlog" exited with code 1.
    0 Warning(s)
    3 Error(s)
```

The cause of this build error appeared in https://github.com/dotnet/runtime/pull/67783:

https://github.com/dotnet/runtime/blob/4ed596ef63e60ce54cfb41d55928f0fe45f65cf3/src/tests/Loader/classloader/generics/ByRefLike/InvalidCSharp.ilproj#L5-L8

This means that during cross build on x64 for arm/armel/arm64 incorrect ilasm binary with non-x64 arch will be tried to be executed. I guess that this is a temporary change according to comment.

cc @alpencolt

--- a/eng/Version.Details.xml
+++ b/eng/Version.Details.xml
@@ -226,13 +226,13 @@
       <Uri>https://github.com/dotnet/runtime</Uri>
       <Sha>b8dc04b445fc98d24fd92262364a9e5b5e23a5b9</Sha>
     </Dependency>
-    <Dependency Name="Microsoft.NETCore.ILAsm" Version="7.0.0-preview.4.22208.8">
+    <Dependency Name="Microsoft.NETCore.ILAsm" Version="7.0.0-preview.5.22226.6">
       <Uri>https://github.com/dotnet/runtime</Uri>
-      <Sha>b8dc04b445fc98d24fd92262364a9e5b5e23a5b9</Sha>
+      <Sha>c10154087e734fa58f08f2797e5d99404e9cd96b</Sha>
     </Dependency>
-    <Dependency Name="Microsoft.NET.Sdk.IL" Version="7.0.0-preview.4.22208.8">
+    <Dependency Name="Microsoft.NET.Sdk.IL" Version="7.0.0-preview.5.22226.6">
       <Uri>https://github.com/dotnet/runtime</Uri>
-      <Sha>b8dc04b445fc98d24fd92262364a9e5b5e23a5b9</Sha>
+      <Sha>c10154087e734fa58f08f2797e5d99404e9cd96b</Sha>
     </Dependency>
     <Dependency Name="System.Text.Json" Version="7.0.0-preview.4.22208.8">
       <Uri>https://github.com/dotnet/runtime</Uri>

--- a/eng/Versions.props
+++ b/eng/Versions.props
@@ -77,7 +77,7 @@
     <MicrosoftNETCoreDotNetHostPolicyVersion>7.0.0-preview.4.22208.8</MicrosoftNETCoreDotNetHostPolicyVersion>
     <MicrosoftExtensionsDependencyModelVersion>3.1.0</MicrosoftExtensionsDependencyModelVersion>
     <!-- CoreClr dependencies -->
-    <MicrosoftNETCoreILAsmVersion>7.0.0-preview.4.22208.8</MicrosoftNETCoreILAsmVersion>
+    <MicrosoftNETCoreILAsmVersion>7.0.0-preview.5.22226.6</MicrosoftNETCoreILAsmVersion>
     <runtimelinuxarm64MicrosoftNETCoreRuntimeObjWriterVersion>1.0.0-alpha.1.22204.1</runtimelinuxarm64MicrosoftNETCoreRuntimeObjWriterVersion>
     <runtimelinuxx64MicrosoftNETCoreRuntimeObjWriterVersion>1.0.0-alpha.1.22204.1</runtimelinuxx64MicrosoftNETCoreRuntimeObjWriterVersion>
     <runtimelinuxmuslarm64MicrosoftNETCoreRuntimeObjWriterVersion>1.0.0-alpha.1.22204.1</runtimelinuxmuslarm64MicrosoftNETCoreRuntimeObjWriterVersion>

--- a/global.json
+++ b/global.json
@@ -13,6 +13,6 @@
     "Microsoft.DotNet.SharedFramework.Sdk": "7.0.0-beta.22179.1",
     "Microsoft.Build.NoTargets": "3.3.0",
     "Microsoft.Build.Traversal": "3.1.6",
-    "Microsoft.NET.Sdk.IL": "7.0.0-preview.4.22208.8"
+    "Microsoft.NET.Sdk.IL": "7.0.0-preview.5.22226.6"
   }
 }

--- a/src/tests/Loader/classloader/generics/ByRefLike/InvalidCSharp.ilproj
+++ b/src/tests/Loader/classloader/generics/ByRefLike/InvalidCSharp.ilproj
@@ -2,15 +2,6 @@
   <PropertyGroup>
     <OutputType>library</OutputType>
     <CLRTestKind>SharedLibrary</CLRTestKind>
-    <!-- Use the locally built ILAsm until the published version supports the new IL keywords
-        When the below is removed, also update the test project to enable ILAsm roundtrip testing.
-      -->
-    <ILAsmToolPath>$(ArtifactsBinDir)/coreclr/$(OSPlatformConfig)</ILAsmToolPath>
-
-    <!-- This override is for the AnyOS build of managed libraries. When this is detected we defer to the host OS value
-        and also to the Release build as that is what is used for the AnyOS build leg.
-      -->
-    <ILAsmToolPath Condition="'$(TargetOS)' == 'AnyOS'">$(ArtifactsBinDir)/coreclr/$(_hostOS).$(TargetArchitecture).Release</ILAsmToolPath>
     <MonoAotIncompatible>true</MonoAotIncompatible>
     <DisableProjectBuild Condition="'$(RuntimeFlavor)' == 'Mono'">true</DisableProjectBuild>
   </PropertyGroup>

--- a/src/tests/Loader/classloader/generics/ByRefLike/Validate.csproj
+++ b/src/tests/Loader/classloader/generics/ByRefLike/Validate.csproj
@@ -1,8 +1,6 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <!-- See comment in InvalidCSharp project about local ILAsm. -->
-    <IlasmRoundTripIncompatible>true</IlasmRoundTripIncompatible>
     <DisableProjectBuild Condition="'$(RuntimeFlavor)' == 'Mono'">true</DisableProjectBuild>
   </PropertyGroup>
   <ItemGroup>

